tests/test_nmapconnector.py

```diff
@@ -0,0 +1,206 @@
+
+
+def test_nmapconnector_scan_ipv4() -> None:
+    """
+    Check if we can successfully scan 127.0.0.1 host
+    :return:
+    """
+    nm.scan(
+        host=NmapTestDefaultValues.HOST4,
+        arguments=DefaultNmapScanValues.ARGUMENTS,
+        ports=str(NmapTestDefaultValues.PORT4),
+    )
+
+    assert (
+        nm.get_results()
+        .get(NmapTestDefaultValues.HOST4)
+        .get("tcp")
+        .get(NmapTestDefaultValues.PORT4, False)
+    )
+
+
+def test_nmapconnector_scan_ipv6() -> None:
+    """
+    Check if we can successfully scan ::1 host
+    :return:
+    """
+    nm.scan(
+        host=NmapTestDefaultValues.HOST6,
+        arguments=DefaultNmapScanValues.ARGUMENTS,
+        ports=str(NmapTestDefaultValues.PORT6),
+    )
+
+    assert (
+        nm.get_results()
+        .get(NmapTestDefaultValues.HOST6)
+        .get("tcp")
+        .get(NmapTestDefaultValues.PORT6, False)
+    )
+
+
+def test_nmapconnector_scan_error() -> None:
+    """
+    Raise NmapConnectorScanError and check output of it
+    :return:
+    """
+    with patch(
+        "grinder.nmapconnector.NmapConnector.scan",
+        side_effect=NmapConnectorScanError("test"),
+    ):
+        with raises(NmapConnectorScanError) as scan_error:
+            nm.scan(host=NmapTestDefaultValues.HOST4)
+        assert "Error occured in Nmap Connector module: test" == str(scan_error.value)
+
+
+def test_nmpaconnector_scan_without_any_args() -> None:
+    """
+    Test NmapConnector scan running without any args
+    :return:
+    """
+    with raises(NmapConnectorScanError) as scan_error:
+        nm.scan(host="", arguments="", ports="", sudo=False)
+    assert "'' does not appear to be an IPv4 or IPv6 address" in str(scan_error.value)
+
+
+def test_nmpaconnector_scan_bad_argument() -> None:
+    """
+    Test NmapConnector scan running with bad arguments
+    :return:
+    """
+    with raises(NmapConnectorScanError) as scan_error:
+        nm.scan(host=NmapTestDefaultValues.HOST4, arguments="--bad-argument")
+    assert "nmap: unrecognized option '--bad-argument'" in str(scan_error.value)
+
+
+def test_nmapconnector_get_results() -> None:
+    """
+    Check if we can successfully get NmapConnector scan results
+    :return:
+    """
+    nm.get_results()
+
+
+def test_nmapconnector_get_results_error() -> None:
+    """
+    Raise NmapConnectorGetResultsError and check output of it
+    :return:
+    """
+    with patch(
+        "grinder.nmapconnector.NmapConnector.get_results",
+        side_effect=NmapConnectorGetResultsError("test"),
+    ):
+        with raises(NmapConnectorGetResultsError) as get_results_error:
+            nm.get_results()
+        assert "Error occured in Nmap Connector module: test" == str(
+            get_results_error.value
+        )
+
+
+def test_nmapconnector_get_results_count() -> None:
+    """
+    Check if we can successfully get NmapConnector scan results
+    :return:
+    """
+    nm.get_results_count()
```

Comment:
The same as for the `test_nmapconnector_get_results()`, check that default value for temporary instance is `0`.

Reply:
Also, if you check that case in `test_nmapconnector_get_results()` with separate function as I said, you need to add separate function for default values here: `test_nmapconnector_get_results_count_default_values()`